src/ncurses_ui.cc

```diff
@@ -612,15 +613,31 @@ Optional<Key> NCursesUI::get_next_key()
                 enable_mouse(mouse_enabled);
                 return {};
             }
-            return ctrl(Codepoint(c) - 1 + 'a');
+            return {ctrl(Codepoint(c) - 1 + 'a')};
         }
 
-        for (int i = 0; i < 12; ++i)
         {
-            if (c == KEY_F(i + 1))
-                return {Key::F1 + i};
-            if (c == KEY_F(m_shift_function_key + i + 1))
-                return shift(Key::F1 + i);
+            KeyList keys;
+            for (int i = 0; i < 12; ++i)
+            {
+                if (c == KEY_F(i + 1))
+                {
+                    keys.emplace_back(Key::F1 + i);
+                    break;
+                }
+            }
+            for (int i = 0; i < 12; ++i)
+            {
+                if (c == KEY_F(m_shift_function_key + i + 1))
```

Comment:
Since this function returns a list now, I think you can do this in a single loop like before:

    for (int i = 0; i < 12; ++i)
    {
        if (c == KEY_F(i + 1)) keys.emplace_back(Key::F1 + i);
        if (c == KEY_F(m_shift_function_key + i + 1)) keys.push_back(shift(Key::F1 + i));
    }

Reply:
Yes, I wasn't sure about this. Let us first see if @mawww agrees with the general approach.